### PR TITLE
Ensure `get_all_stub_files()` yields modules in a deterministic order

### PR DESCRIPTION
**Pull request by AlexWaygood**
While working on https://github.com/JelleZijlstra/stubdefaulter/pull/16, I was comparing the terminal output of `stubdefaulter` with and without the change in order to test my PR. This was made slightly harder by the fact that `stubdefaulter` wasn't processing typeshed's stdlib in a completely deterministic order, which surprised me.

The root cause is that `typeshed_client.finder.get_all_stub_files()` does not yield stub files in a deterministic order. This PR is an attempt to fix that.

I'm not overly attached to this PR -- the problem it's solving isn't a particularly serious one -- so I won't be particularly upset if you don't like it :) I _think_ this patch makes the output deterministic (I've done manual testing), but I've failed to come up with a way to test this in CI.

### CODE REVIEW

File: typeshed_client/finder.py [hunk 1]
```diff
@@ -175,9 +175,11 @@ def _get_all_stub_files_from_directory(
     directory: _DirEntry, root_directory: Path, seen: Set[str]
 ) -> Generator[Tuple[str, Path], None, Set[str]]:
     new_seen = set(seen)
-    to_do: Set[os.PathLike[str]] = {directory}
+    # Use a dict for to_do to ensure modules are yielded in a deterministic order
+    # The keys in the dict are irrelevant; this is just a standin for an ordered set
```
Comment (JelleZijlstra) [inline typeshed_client/finder.py:179]:
```suggestion
    # The values in the dict are irrelevant; this is just a standin for an ordered set
```

File: typeshed_client/finder.py [hunk 2]
```diff
@@ -175,9 +175,11 @@ def _get_all_stub_files_from_directory(
...
+    to_do: Dict[os.PathLike[str], None] = {directory: None}
```
Comment (JelleZijlstra) [inline typeshed_client/finder.py:180]: I wonder if we could just use a list. Reading the code I'm not sure there's a way for the same file to appear multiple times.

**Comment by JelleZijlstra** [comment]
Also seems like I need to drop support for 3.6 :)

File: typeshed_client/finder.py (hunk 2 above)
Comment (AlexWaygood) [reply typeshed_client/finder.py:180]: I assumed that there was a way for duplicates to creep in due to the use of a set currently, but I agree that a list would certainly be cleaner :)

Comment (AlexWaygood) [reply typeshed_client/finder.py]: Updated the PR to use a list

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**